packages/admin/admin/src/common/buttons/feedback/FeedbackButton.tsx

```diff
@@ -0,0 +1,181 @@
+import { ThreeDotSaving } from "@comet/admin-icons";
+import { Button as MuiButton, ButtonClassKey, ButtonProps, ComponentsOverrides } from "@mui/material";
+import { styled, Theme, useThemeProps } from "@mui/material/styles";
+import * as React from "react";
+
+import { Tooltip, TooltipClassKey } from "../../Tooltip";
+
+export type FeedbackButtonClassKey = "idle" | "loading" | "success" | "fail" | ButtonClassKey;
+
+type OwnerState = Pick<FeedbackButtonProps, "variant" | "color"> & { displayState?: FeedbackButtonDisplayState };
+
+const Button = styled(MuiButton, {
+    name: "CometAdminFeedbackButton",
+    slot: "button",
+    overridesResolver({ ownerState }: { ownerState: OwnerState }, styles) {
+        return [
+            styles.root,
+            ownerState.displayState === "idle" && styles.idle,
+            ownerState.displayState === "fail" && styles.fail,
+            ownerState.displayState === "success" && styles.success,
+            ownerState.displayState === "loading" && styles.loading,
+        ];
+    },
+})<{ ownerState: OwnerState }>();
+
+export interface FeedbackButtonProps extends ButtonProps {
+    loading?: boolean;
+    hasErrors?: boolean;
+    loadingIcon?: React.ReactNode;
+    startIcon?: React.ReactNode;
+    endIcon?: React.ReactNode;
+    tooltipSuccessMessage?: string;
+    tooltipErrorMessage?: string;
+}
+
+export type FeedbackButtonDisplayState = "idle" | "loading" | "success" | "fail";
+
+export function FeedbackButton(inProps: FeedbackButtonProps) {
+    const {
+        loading = false,
+        hasErrors = false,
```

Comment:
`false` as a default-value for props should not be necessary. 

```suggestion
        loading,
        hasErrors,
```